closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -27,13 +27,108 @@
 #include <openssl/err.h>
 #include <openssl/rsa.h>
 
+#include <assert.h>
 #include <jni.h>
 #include <stdio.h>
+#include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
-#include <assert.h>
 
 #include "jdk_crypto_jniprovider_NativeCrypto.h"
+#include "NativeCrypto_md.h"
+
+
+//Header for RSA algorithm using 1.0.2 OpenSSL
+int OSSL102_RSA_set0_key(RSA *, BIGNUM *, BIGNUM *, BIGNUM *);
+int OSSL102_RSA_set0_factors(RSA *, BIGNUM *, BIGNUM *);
+int OSSL102_RSA_set0_crt_params(RSA *, BIGNUM *, BIGNUM *, BIGNUM *);
+
+//Type definitions of function pointers
```

Comment:
The following typedefs are **not** function _pointers_. This is how you would declare the first pointer:
```
typedef char * (*OSSL_error_string_t)(unsigned long, char *);
```

Reply:
Keith, this is the way Nasser did this and he created a pointer to the functions later in the code. 
```
//Define pointers for OpenSSL functions to handle Errors.
OSSL_error_string_t* OSSL_error_string;
OSSL_get_error_t* OSSL_get_error;
```
Are you saying that this is wrong because that is what the comment is saying. 
If so, we can change the comment.

Reply:
It's a little disconcerting to read comments that suggest the author doesn't understand what they're writing. If I remember correctly, I asked Nasser to change this in jdk8. I can accept it for now, but I think it should be cleaned up eventually.

Reply:
I dont know whether the original author understood what they were writing. I have just created this PR from the code in Nasser's PR and I assumed Nasser understood what he was writing. I do agree that I am not that familiar with the code, not being the original author. I had assumed the code works and from the build/test it does seems to. If you think this is not the right way for this code to be written and should it be cleaned up and then I am sure you can do this via a new issue once this code is merged.